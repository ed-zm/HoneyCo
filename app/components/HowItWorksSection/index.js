import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Container } from 'reactstrap'
import { HoneyCoDarkGray, HoneyCoLightGray } from '../../constants'

const StyledContainer = styled(Container)`
  padding-top: 100px;
`

const StyledBackground = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-flow: row wrap;
`

const StyledBoxContainer = styled.div`
  height: 650px;
  width: 30%;
  display: flex;
  justify-content: space-around;
  flex-flow: row wrap;
`

const StyledBox = styled.div`
  color: ${HoneyCoDarkGray};
  width: 100%;
  height: 250px;
  background-color: ${HoneyCoLightGray};
  padding: 1em;
`

const StyledBackgroundContainer = styled.div`
  height: 650px;
  width: 30%;
  background: url(${(props) => props.background});
  background-position: center;
  background-size: cover;
`

const StyledTitle = styled.span`
  font-size: 70px;
  display: block;
  width: 100%;
  text-align: center;
  color: ${HoneyCoDarkGray};
`
const StyledBoxTitle = styled.span`
  font-size: 32px;
  display: block;
  margin-bottom: 1em;
`
const StyledBoxDescription = styled.span`
  font-size: 20px;
  display: block;
  
`

const HowItWorksSection = (props) =>
<StyledContainer>
  <StyledBackground>
    <StyledTitle>Your 24/7 Piece of Mind</StyledTitle>
    <StyledBoxContainer>
      <StyledBox>
        <StyledBoxTitle>{props.text.box1.title}</StyledBoxTitle>
        <StyledBoxDescription>{props.text.box1.description}</StyledBoxDescription>
      </StyledBox>
      <StyledBox>
        <StyledBoxTitle>{props.text.box2.title}</StyledBoxTitle>
        <StyledBoxDescription>{props.text.box2.description}</StyledBoxDescription>
      </StyledBox>
    </StyledBoxContainer>
    <StyledBackgroundContainer background = {props.background}/>
    <StyledBoxContainer>
      <StyledBox>
        <StyledBoxTitle>{props.text.box3.title}</StyledBoxTitle>
        <StyledBoxDescription>{props.text.box3.description}</StyledBoxDescription>
      </StyledBox><StyledBox>
        <StyledBoxTitle>{props.text.box4.title}</StyledBoxTitle>
        <StyledBoxDescription>{props.text.box4.description}</StyledBoxDescription>
      </StyledBox>
    </StyledBoxContainer>
  </StyledBackground>
</StyledContainer>

HowItWorksSection.propTypes = {
  text: PropTypes.object.isRequired,
}
export default HowItWorksSection
