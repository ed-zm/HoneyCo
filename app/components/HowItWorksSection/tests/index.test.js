import React from 'react'
import { shallow } from 'enzyme'
import HowItWorksSection from '../index'
const text = {
  box1: {
    title: 'title1',
    description: 'description1',
  },
  box2: {
    title: 'title2',
    description: 'description2',
  },
  box3: {
    title: 'title3',
    description: 'description3',
  },
  box4: {
    title: 'title4',
    description: 'description4',
  },
}

describe('<HowItWorksSection />', () => {
  it('It should render the HowItWorksSection Component', () => {
    const wrapper = shallow(<HowItWorksSection text={text}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
