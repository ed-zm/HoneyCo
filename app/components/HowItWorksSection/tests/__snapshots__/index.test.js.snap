// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HowItWorksSection /> It should render the HowItWorksSection Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(gt)>
    <styled.div>
        <styled.span>
            Your 24/7 Piece of Mind
        </styled.span>
        <styled.div>
            <styled.div>
                <styled.span>
                    title1
                </styled.span>
                <styled.span>
                    description1
                </styled.span>
            </styled.div>
            <styled.div>
                <styled.span>
                    title2
                </styled.span>
                <styled.span>
                    description2
                </styled.span>
            </styled.div>
        </styled.div>
        <styled.div
            background={undefined}
        />
        <styled.div>
            <styled.div>
                <styled.span>
                    title3
                </styled.span>
                <styled.span>
                    description3
                </styled.span>
            </styled.div>
            <styled.div>
                <styled.span>
                    title4
                </styled.span>
                <styled.span>
                    description4
                </styled.span>
            </styled.div>
        </styled.div>
    </styled.div>
</Styled(gt)>,
  "nodes": Array [
    <Styled(gt)>
      <styled.div>
            <styled.span>
                  Your 24/7 Piece of Mind
            </styled.span>
            <styled.div>
                  <styled.div>
                        <styled.span>
                              title1
                        </styled.span>
                        <styled.span>
                              description1
                        </styled.span>
                  </styled.div>
                  <styled.div>
                        <styled.span>
                              title2
                        </styled.span>
                        <styled.span>
                              description2
                        </styled.span>
                  </styled.div>
            </styled.div>
            <styled.div
                  background={undefined}
            />
            <styled.div>
                  <styled.div>
                        <styled.span>
                              title3
                        </styled.span>
                        <styled.span>
                              description3
                        </styled.span>
                  </styled.div>
                  <styled.div>
                        <styled.span>
                              title4
                        </styled.span>
                        <styled.span>
                              description4
                        </styled.span>
                  </styled.div>
            </styled.div>
      </styled.div>
</Styled(gt)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <HowItWorksSection
        text={
                Object {
                        "box1": Object {
                          "description": "description1",
                          "title": "title1",
                        },
                        "box2": Object {
                          "description": "description2",
                          "title": "title2",
                        },
                        "box3": Object {
                          "description": "description3",
                          "title": "title3",
                        },
                        "box4": Object {
                          "description": "description4",
                          "title": "title4",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "text": Object {
            "box1": Object {
              "description": "description1",
              "title": "title1",
            },
            "box2": Object {
              "description": "description2",
              "title": "title2",
            },
            "box3": Object {
              "description": "description3",
              "title": "title3",
            },
            "box4": Object {
              "description": "description4",
              "title": "title4",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(gt)>
          <styled.div>
                    <styled.span>
                              Your 24/7 Piece of Mind
                    </styled.span>
                    <styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title1
                                        </styled.span>
                                        <styled.span>
                                                  description1
                                        </styled.span>
                              </styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title2
                                        </styled.span>
                                        <styled.span>
                                                  description2
                                        </styled.span>
                              </styled.div>
                    </styled.div>
                    <styled.div
                              background={undefined}
                    />
                    <styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title3
                                        </styled.span>
                                        <styled.span>
                                                  description3
                                        </styled.span>
                              </styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title4
                                        </styled.span>
                                        <styled.span>
                                                  description4
                                        </styled.span>
                              </styled.div>
                    </styled.div>
          </styled.div>
</Styled(gt)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(gt)>
          <styled.div>
                    <styled.span>
                              Your 24/7 Piece of Mind
                    </styled.span>
                    <styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title1
                                        </styled.span>
                                        <styled.span>
                                                  description1
                                        </styled.span>
                              </styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title2
                                        </styled.span>
                                        <styled.span>
                                                  description2
                                        </styled.span>
                              </styled.div>
                    </styled.div>
                    <styled.div
                              background={undefined}
                    />
                    <styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title3
                                        </styled.span>
                                        <styled.span>
                                                  description3
                                        </styled.span>
                              </styled.div>
                              <styled.div>
                                        <styled.span>
                                                  title4
                                        </styled.span>
                                        <styled.span>
                                                  description4
                                        </styled.span>
                              </styled.div>
                    </styled.div>
          </styled.div>
</Styled(gt)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HowItWorksSection
    text={
        Object {
            "box1": Object {
              "description": "description1",
              "title": "title1",
            },
            "box2": Object {
              "description": "description2",
              "title": "title2",
            },
            "box3": Object {
              "description": "description3",
              "title": "title3",
            },
            "box4": Object {
              "description": "description4",
              "title": "title4",
            },
          }
    }
/>,
}
`;
